pkg/api/customization/pipeline/pipeline.go

```diff
@@ -0,0 +1,141 @@
+	logrus.Debugf("do pipeline action:%s", actionName)
+
+	switch actionName {
+	case "activate":
+		return h.activate(apiContext)
+	case "deactivate":
+		return h.deactivate(apiContext)
+	case "run":
+		return h.run(apiContext)
+	case "export":
+		return h.export(apiContext)
+	}
+	return httperror.NewAPIError(httperror.InvalidAction, "unsupported action")
+}
+
+func (h *Handler) activate(apiContext *types.APIContext) error {
+	parts := strings.Split(apiContext.ID, ":")
+	ns := parts[0]
+	name := parts[1]
+
+	pipelines := h.Management.Management.Pipelines(ns)
+	pipeline, err := pipelines.Get(name, v1.GetOptions{})
+	if err != nil {
+		logrus.Errorf("Error while getting pipeline for %s :%v", apiContext.ID, err)
+		return err
+	}
+
+	if pipeline.Status.State != "active" {
+		pipeline.Status.State = "active"
+		if _, err = pipelines.Update(pipeline); err != nil {
+			logrus.Errorf("Error while updating pipeline:%v", err)
+			return err
+		}
+	}
+
+	data := map[string]interface{}{}
+	if err := access.ByID(apiContext, apiContext.Version, apiContext.Type, apiContext.ID, &data); err != nil {
+		return err
+	}
+
+	apiContext.WriteResponse(http.StatusOK, data)
+	return nil
+}
+
+func (h *Handler) deactivate(apiContext *types.APIContext) error {
+	parts := strings.Split(apiContext.ID, ":")
+	ns := parts[0]
+	name := parts[1]
+
+	pipelines := h.Management.Management.Pipelines(ns)
+	pipeline, err := pipelines.Get(name, v1.GetOptions{})
+	if err != nil {
+		logrus.Errorf("Error while getting pipeline for %s :%v", apiContext.ID, err)
+		return err
+	}
+
+	if pipeline.Status.State != "inactive" {
+		pipeline.Status.State = "inactive"
+		if _, err = pipelines.Update(pipeline); err != nil {
+			logrus.Errorf("Error while updating pipeline:%v", err)
+			return err
+		}
+	}
+
+	data := map[string]interface{}{}
+	if err := access.ByID(apiContext, apiContext.Version, apiContext.Type, apiContext.ID, &data); err != nil {
+		return err
+	}
+
+	apiContext.WriteResponse(http.StatusOK, data)
+	return nil
+}
+
+func (h *Handler) run(apiContext *types.APIContext) error {
+	parts := strings.Split(apiContext.ID, ":")
+	ns := parts[0]
+	name := parts[1]
+	pipelines := h.Management.Management.Pipelines(ns)
+
+	executions := h.Management.Management.PipelineExecutions(ns)
+	logs := h.Management.Management.PipelineExecutionLogs(ns)
+
+	pipeline, err := pipelines.Get(name, v1.GetOptions{})
+	if err != nil {
+		logrus.Errorf("Error while getting pipeline for %s :%v", apiContext.ID, err)
+		return err
+	}
+	execution, err := utils.RunPipeline(pipelines, executions, logs, pipeline, utils.TriggerTypeUser)
+	if err != nil {
+		return err
+	}
+
+	data := map[string]interface{}{}
+	if err := access.ByID(apiContext, apiContext.Version, client.PipelineExecutionType, ns+":"+execution.Name, &data); err != nil {
+		return err
+	}
+
+	apiContext.WriteResponse(http.StatusOK, data)
+	return err
+}
+
+func (h *Handler) export(apiContext *types.APIContext) error {
+	return nil
```

Comment:
there is no implementation for export action? If so, should we disable it for now?